src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/filter/FilterClause.java

```diff
@@ -19,7 +19,7 @@
     example = """
              {"name": "Aaron", "country": "US"}
               """)
-public record FilterClause(LogicalExpression logicalExpression) {
```

Comment:
Does this need to be hidden to prevent cyclic definition or... ?

Reply:
Exactly

Reply:
Will it have any negative impact? (like not exposing our API schema as intended?)

Reply:
```
      "FilterClause": {
        "type": "object",
        "properties": {
          "logicalExpression": {
            "$ref": "#/components/schemas/LogicalExpression"
          }
        },
        "example": {
          "name": "Aaron",
          "country": "US"
        }
      },
```

For example, previous we have FilterClause schema like this.
Now it will be 

  ```
    "FilterClause" : {
        "type" : "object",
        "example" : {
          "name" : "Aaron",
          "country" : "US"
        }
      },
```

Reply:
Is there a need to expose logicalExpression property?

Reply:
I believe yes, since its part of the API definition. I see it is used in `DeleteManyCommand` and `DeleteOneCommand` as well apart from the `FilterClause`.

Reply:
Sure, let's discuss it.
So, we certainly has the circular reference in our design for logicalExpression, that is for the logical operator like $or and $and.

Regarding the openAPI world, the criteria for the circular reference is different. For example, swaggerUI will work with a circular reference spec. However, for example, GPT action validator will error with a circular reference spec.

Some references: 
https://apitools.dev/swagger-parser/docs/#:~:text=//%20Error%0A%7D-,Circular%20%24Refs,-Swagger%20APIs%20can
https://github.com/stoplightio/prism/issues/1456

Reply:
👍 I understand that part. I'm not sure if hiding that field from the schema is the solution here. Should we work with the OpenAI team to understand on how this kind of schema can be validated (because having recursive seems in a OpenAPI spec seems normal to me).